src/System.Private.CoreLib/src/System/Utf8String.Construction.cs

```diff
@@ -159,47 +233,437 @@ private unsafe Utf8String Ctor(char* value)
+            // contain well-formed UTF-16 data. This method could fail in other ways, such as
+            // throwing an OutOfMemoryException if allocation of the output parameter fails.
+
+            value = CreateFromUtf16Common(buffer, replaceInvalidSequences: false);
+            return !(value is null);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="Utf8String"/> instance from existing UTF-8 data.
+        /// </summary>
+        /// <param name="buffer">The existing data from which to create the new <see cref="Utf8String"/>.</param>
+        /// <remarks>
+        /// If <paramref name="buffer"/> contains any ill-formed UTF-8 subsequences, those subsequences will
+        /// be replaced with <see cref="Rune.ReplacementChar"/> in the returned <see cref="Utf8String"/> instance.
+        /// This may result in the returned <see cref="Utf8String"/> having different contents (and thus a different
+        /// total byte length) than the source parameter <paramref name="buffer"/>.
+        /// </remarks>
+        public static Utf8String CreateFromRelaxed(ReadOnlySpan<byte> buffer)
+        {
+            if (buffer.IsEmpty)
+            {
+                return Empty;
+            }
+
+            // Create and populate the Utf8String instance.
+
+            Utf8String newString = FastAllocateSkipZeroInit(buffer.Length);
+            Buffer.Memmove(ref newString.DangerousGetMutableReference(), ref MemoryMarshal.GetReference(buffer), (uint)buffer.Length);
+
+            // Now perform validation & fixup.
+
+            return Utf8Utility.ValidateAndFixupUtf8String(newString);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="Utf8String"/> instance from existing UTF-16 data.
+        /// </summary>
+        /// <param name="buffer">The existing data from which to create the new <see cref="Utf8String"/>.</param>
+        /// <remarks>
+        /// If <paramref name="buffer"/> contains any ill-formed UTF-16 subsequences, those subsequences will
+        /// be replaced with <see cref="Rune.ReplacementChar"/> in the returned <see cref="Utf8String"/> instance.
+        /// This may result in the original string data not round-tripping properly; that is, calling
+        /// <see cref="ToString"/> on the returned <see cref="Utf8String"/> instance may produce a <see cref="string"/>
+        /// whose contents differ from <paramref name="buffer"/>.
+        /// </remarks>
+        public static Utf8String CreateFromRelaxed(ReadOnlySpan<char> buffer)
+        {
+            Utf8String? newString = CreateFromUtf16Common(buffer, replaceInvalidSequences: true);
+
+            if (newString is null)
+            {
+                // This shouldn't happen unless somebody mutated the input buffer in the middle
+                // of data processing. We just fail in this scenario rather than retrying.
+
+                throw new ArgumentException(
```

Comment:
Same (throw helper) here. Though according to the code-comment, this should be rare, so throw helper has no real benefit, except consistency with other places.
(I would keep the code here as is).